src/components/DeleteEditButtons/index.js

```diff
@@ -2,103 +2,75 @@ import React from 'react';
 import { Link } from 'react-router-dom';
 
-import {
-  Button,
-  Dialog,
-  DialogActions,
-  DialogContent,
-  DialogContentText,
-  DialogTitle,
-} from '@material-ui/core';
+import { Button } from '@material-ui/core';
 import { withStyles } from '@material-ui/core/styles';
 import DeleteIcon from '@material-ui/icons/Delete';
 import EditIcon from '@material-ui/icons/Edit';
 import styles from './styles';
 
 import * as ROUTES from '../../constants/routes';
 import { deleteEventById } from '../../services/events';
+import { ConfirmModal } from "../modals";
 
-const INITIAL_STATES = {
-  isConfirmationModalOpen: false,
-};
-
-class DeleteEditButtons extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { ...INITIAL_STATES };
-  }
-
-  handleModalOpen = () => {
-    this.setState({ isConfirmationModalOpen: true });
-  };
-
-  handleModalClose = () => {
-    this.setState({ isConfirmationModalOpen: false });
-  };
+const DeleteEditButtons = props => {
+  const { classes, eventId } = props;
 
-  handleDeleteEvent = eventId => {
-    deleteEventById(eventId)
+  const handleDeleteEvent = eventToDeleteId => {
+    deleteEventById(eventToDeleteId)
       .then(res => {
         return res;
       })
       .catch(err => {
         console.log(err);
       });
+  };
 
-    this.handleModalClose();
+  const openButtonProps = {
+    openButtonName: 'Delete',
+    openButtonOnClick: null,
+    className: classes.delete,
+    startIcon: <DeleteIcon />,
+    primary: true,
+    negative: true,
   };
 
-  render() {
-    const { classes, eventId } = this.props;
-    const { isConfirmationModalOpen } = this.state;
+  const confirmButtonProps = {
+    buttonName: 'Yes',
+    onClick: () => handleDeleteEvent(eventId),
+    to: ROUTES.CALENDAR,
+    component: Link,
+    positive: true,
+  };
 
-    return (
-      <div className={classes.root}>
-        <Button
-          className={classes.delete}
-          variant='contained'
-          color='secondary'
-          onClick={this.handleModalOpen}
-          startIcon={<DeleteIcon />}
-        >
-          Delete
-        </Button>
-        <Dialog open={isConfirmationModalOpen} onClose={this.handleModalClose}>
-          <DialogTitle>Delete this event?</DialogTitle>
-          <DialogContent>
-            <DialogContentText>
-              Do you want to delete this event completely?
-            </DialogContentText>
-          </DialogContent>
-          <DialogActions>
-            <Button
-              color='primary'
-              to={ROUTES.CALENDAR}
-              component={Link}
-              onClick={() => this.handleDeleteEvent(eventId)}
-            >
-              Yes
-            </Button>
-            <Button onClick={this.handleModalClose} color='primary'>
-              No
-            </Button>
-          </DialogActions>
-        </Dialog>
+  const cancelButtonProps = {
+    buttonName: 'No',
+    onClick: null,
+    positive: true,
```

Comment:
cancel should be negative secondary here I think

Reply:
and we could also just hardcode the cancel button as not doing anything with buttonName as cancel and encapsulate it within the base modal

Reply:
I think it's better to give the user the choice to make their own cancel button, since they might want to put another name other than "cancel", since another name might be more appropriate for their use case. What do you think?

Reply:
we can add it when it comes; my guess is that we don't have to expose that for the next year :')

Reply:
I mean we can just make add the cancel button in ConfirmModal and AlertModal or smth (since they are made from BaseModal).

Reply:
ah yeah that lines up nicely with the idea I was going with in render props - great point!